evaluation/data/cmc/sorc/regrid_CMCsnow_to1Dvector.f90

```diff
@@ -0,0 +1,349 @@
+  status = nf90_get_var(ncid, varid , destination_lookup)
+    if (status /= nf90_noerr) call handle_err(status)
+  
+  status = nf90_inq_varid(ncid, "S", varid)
+    if (status /= nf90_noerr) call handle_err(status)
+  status = nf90_get_var(ncid, varid , weights)
+    if (status /= nf90_noerr) call handle_err(status)
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+! Read lattitude and longitude data
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+   status = nf90_inq_varid(ncid, "yc_b", varid)
+    if (status /= nf90_noerr) call handle_err(status)
+  status = nf90_get_var(ncid, varid , ycb)
+    if (status /= nf90_noerr) call handle_err(status)
+
+  status = nf90_inq_varid(ncid, "xc_b", varid)
+    if (status /= nf90_noerr) call handle_err(status)
+  status = nf90_get_var(ncid, varid , xcb)
+    if (status /= nf90_noerr) call handle_err(status)
+
+  status = nf90_close(ncid)
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+! Read year and numbers of days for a given year
+  open(20, file='year_days.txt', status='old')
+  read(20,*) year, ndays
+  close(20)
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+! Read source data in ascii format
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  source_filename=trim(source_path)//'cmc_sdepth_dly_'//year//'_v01.2.txt'
+  open(10, file=trim(source_filename), status='old')
+  do id = 1, ndays 
+
+  read(10,*) idate ! date in format YYYYMMDDHH where HH is always 00 
+  destination_filename=trim(destination_path)//trim(fv3_grid)//'_cmc_snow_depth_'//idate//'.nc'
+  
+  yyyy=idate(1:4)
+  mm=idate(5:6)
+  dd=idate(7:8)
+
+  write(*,*) yyyy, mm, dd
+  current_date=yyyy//'-'//mm//'-'//dd//' 00:00:00'
+  call calc_sec_since(since_date, current_date, offset_ss, sec_since)
+
+  do i=1,706
+  read(10,'(706(1x,f5.1))') (snowd(i,j),j=1,706) !cm 
+  enddo
+!  write(*,'(706(1x,f5.1))') snowd
+!  stop
+!!!!!!!!!!!! convert cm into mm !!!!!!!!!!!!!!!!!!!!!!!!!!
+  do i=1,706
+   do j=1,706
+   source_input(i,j)=snowd(i,j)*10.0
+   end do
+  end do
+  !write(*,*) source_input
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+! Regrid the data
+  cmcSnowDepth = 0.0
+  cmcSnowDepthWT = 0.0
+
+  do iwt = 1, weight_locs
+
+    latloc = source_lookup(iwt)/source_lons + 1
+    lonloc = source_lookup(iwt) - (latloc-1)*source_lons
+       
+    if(source_input(lonloc,latloc) >= 0.0 ) then
+    cmcSnowDepth(destination_lookup(iwt)) = cmcSnowDepth(destination_lookup(iwt)) + weights(iwt) * source_input(lonloc,latloc)
+    cmcSnowDepthWT(destination_lookup(iwt)) = cmcSnowDepthWT(destination_lookup(iwt)) + weights(iwt)
+    end if
+
+  end do
+  
+  do iwt = 1, destination_locs
+
+    if(cmcSnowDepthWT(iwt) > 0.0 ) then
+     cmcSnowDepth(iwt) = cmcSnowDepth(iwt)/cmcSnowDepthWT(iwt)
+    else
+     cmcSnowDepth(iwt) = -9999.0
+    end if
+
+  end do
+  !write(*,'(192(1x,f5.1))') cmcSnowDepth 
+
+  do iwt = 1, destination_locs
```

Comment:
what is the purpose of this? put a comment above this section so that we know the reason

Reply:
I just used this code, could you please give me 1-2 sentences to describe the purpose? Thank you.